### ISSUE TITLE
feature request: eslint-disable comments are time-boxed (snoozeable)

### ISSUE THREAD

**Issue by dancingshell** [eslint/eslint#13651]
**The version of ESLint you are using.**
`7.5.0`

**The problem you want to solve.**
We recently rolled out mandatory eslint checks as part of our CI for all new code changed. However we would like some flexibility to disable (ignore) these rules. Some example where someone might want to ignore rules
- someone needs to make a hot fix on a piece of code they don't maintain
- someone needs to revert a PR with previously unlinted code
- a QA needs to add some metadata for automation like a qa-tag but doesn't want (or know how) to fix something like cyclomatic complexity when modifying these files

Currently, the only way to ignore code in those cases is to either `eslint-disable` the whole file or just the block/rule that applies if possible. However, this disable comment is unlikely to be removed/discovered by later developers who have the time/knowledge to implement the correct style or quality fix. The code is more likely to remain as is once it is ignored.

**Your take on the correct solution to problem.**
Add a date to the eslint disable comments. I am calling this "snooze" functionality. For ease I will use the whole file as an example, but I think this should be applicable to all rules at all scopes

For example if today is `2020-09-02` and I need to ship a feature I want the option to disable the file today, but tell eslint the date to re-enable that rule in the future.

If I know the change will ship today, I can set the snooze date to tomorrow. All code merged today will pass the eslint check but any code merged tomorrow and after will fail again.
```js
/** eslint-disable 2020-09-03 **/
```

If this code might take a few days to go through CR I can extend the snooze to a week
```js
/** eslint-disable 2020-09-10 **/
```

Any code merged on or after the snooze date will act as if that disable comment doesn't exist. This will allow certain features to ship without passing lint, but avoids the issue of blocks of code or files being forgotten indefinitely.

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by mdjermanovic** [comment]
Hi @dancingshell, thanks for the issue!

This might be a too specific use case to be handled in ESLint core. On the other hand, there's a generic issue about TODO/FIXME work being forgotten indefinitely, so it might be more useful to handle it in a way that doesn't have to be specifically related to eslint-disable comments.

Have you checked this rule from [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn):

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/expiring-todo-comments.md

ESLint comments can have a description after `--`, so for example, if someone wants to disable the `complexity` rule in a file just for a few days, they can write:

```js
/* eslint-disable complexity -- FIXME [2020-09-05] */
```
and then, if the linting issue isn't fixed in time, the `unicorn/expiring-todo-comments` rule will warn on `2020-09-06`.

An additional benefit is that you can add a date to any comment, unrelated to the eslint-disable use case.

A problem might be `/* eslint-disable */` without specified rules, because it disables all rules (including this one) starting from the beginning of the comment. This could be solved by adding the comment above `/* eslint-disable */`:

```js
// FIXME [2020-09-01]
/* eslint-disable */
```

There are also some useful rules about eslint comments in [eslint-plugin-eslint-comments](https://mysticatea.github.io/eslint-plugin-eslint-comments/).
